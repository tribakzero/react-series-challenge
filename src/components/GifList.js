import React, { Component } from 'react';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import favoriteActions from '../actions/favorite';
import Gif from './Gif';

class GifList extends Component {
  toggleFavorite = (gif) => {
    let gifData = (this.props.favorite[gif]) ||Â this.props.gifs[gif]
    if(gifData.liked) {
      this.props.removeFavorite({ gif: gifData });
    } else {
      this.props.addFavorite({ gif: gifData });
    }
  }

  render(){
    return (
      <React.Fragment>
        {
          Object.keys(this.props.gifs).map((gif) =>
            <Gif
              key={this.props.gifs[gif].id}
              id={this.props.gifs[gif].id}
              liked={this.props.gifs[gif].liked}
              src={this.props.gifs[gif].images.fixed_width_small.url}
              onClick={() => this.toggleFavorite(gif)}
            />
          )
        }
      </React.Fragment>
    )
  }
}



const mapStateToProps = (state) => {
  const {
    favorite
  } = state;
  
  return {
    favorite
  };
};

const mapDispatchToProps = (dispatch) => {
  const { removeFavorite, addFavorite } = favoriteActions.creators;

  return bindActionCreators({
    removeFavorite,
    addFavorite
  }, dispatch);
};

export default connect(mapStateToProps, mapDispatchToProps)(GifList);
