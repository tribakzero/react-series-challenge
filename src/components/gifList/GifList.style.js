import styled from 'styled-components';

export const StyledGifList = styled.div`
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
  .gif {
    width: auto;
    margin: 1em;
  }
`
