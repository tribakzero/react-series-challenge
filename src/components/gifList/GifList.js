import React from 'react';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import favoriteActions from 'actions/favorite';
import Gif from 'components/gif';
import { StyledGifList } from './GifList.style'

const GifList = (props) => {
  let toggleFavorite = (gif) => {
    let gifData = (props.favorite[gif]) ||Â props.gifs[gif]
    if(gifData.liked) {
      props.removeFavorite({ gif: gifData });
    } else {
      props.addFavorite({ gif: gifData });
    }
  }

  return (
    <StyledGifList>
      {
        Object.keys(props.gifs).map((gif) =>
          <Gif
            key={props.gifs[gif].id}
            id={props.gifs[gif].id}
            liked={props.gifs[gif].liked}
            title={props.gifs[gif].title}
            src={props.gifs[gif].images.original.webp}
            onClick={() => toggleFavorite(gif)}
          />
        )
      }
    </StyledGifList>
  )
}



const mapStateToProps = (state) => {
  const {
    favorite
  } = state;
  
  return {
    favorite
  };
};

const mapDispatchToProps = (dispatch) => {
  const { removeFavorite, addFavorite } = favoriteActions.creators;

  return bindActionCreators({
    removeFavorite,
    addFavorite
  }, dispatch);
};

export default connect(mapStateToProps, mapDispatchToProps)(GifList);
