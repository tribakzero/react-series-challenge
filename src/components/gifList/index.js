import GifList from './GifList';

export default GifList;
