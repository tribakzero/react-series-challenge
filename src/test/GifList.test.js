import React from 'react';
import { shallow } from 'enzyme';
import { GifList } from 'components/gifList/GifList'

let wrapper;

beforeEach(() => {
  wrapper = shallow(<GifList/>)
})

describe('GifList', () => {
  test('renders properly', () => {
    expect(wrapper).toMatchSnapshot()
  })
});
