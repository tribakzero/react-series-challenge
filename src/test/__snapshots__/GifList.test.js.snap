// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifList renders properly 1`] = `undefined`;
